endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+        http.Error(w, "Error generating random uuid", http.StatusInternalServerError)
+        return
+    }
+
+    survey.Reference = surveyRef.String()
+
+    _, err = stmt.Exec(survey.Reference, survey.ShortName, survey.LongName, survey.LegalBasis, survey.SurveyMode)
+    if err != nil {
+        http.Error(w, "SQL statement error", http.StatusInternalServerError)
+        return
+    }
+
+    var js []byte
+    js, err = json.Marshal(&survey)
+
+    logger.Logger.Info("Successfully posted survey")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusCreated)
+    w.Write(js)
+}
+
+//Create survey based on JSON request
+func getSurveyByRef (w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    vars := mux.Vars(r)
+
+    surveyRef := vars["reference"]
+
+    queryString := "SELECT survey_ref, short_name, long_name, legal_basis, survey_mode FROM surveyv2.survey WHERE survey_ref = $1"
+
+    rows, err := db.Query(queryString, surveyRef)
+
+    if err != nil {
+        http.Error(w, "get survey query failed", http.StatusInternalServerError)
+        return
+    }
+
+    var listOfSurveys = []models.Survey{}
+
+    for rows.Next(){
+        survey := models.Survey{}
+
+        rows.Scan(
+            &survey.Reference,
+            &survey.ShortName,
+            &survey.LongName,
+            &survey.LegalBasis,
+            &survey.SurveyMode,
+        )
+
+        listOfSurveys = append(listOfSurveys, survey)
+    }
+
+    if len(listOfSurveys) == 0 {
+        re := models.NewRESTError("404", "Survey not found")
+        data, err := json.Marshal(re)
+        if err != nil {
+            http.Error(w, "Error marshaling NewRestError JSON", http.StatusInternalServerError)
+            return
+        }
+
+        w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+        w.WriteHeader(http.StatusNotFound)
+        w.Write(data)
+
+        return
+    }
+
+    data, err := json.Marshal(listOfSurveys)
+    if err != nil {
+        http.Error(w, "Failed to marshal survey JSON", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully retrieved survey from reference")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusOK)
+    w.Write(data)
+
+}
+
+//Delete survey based on given reference
+func deleteSurveyByRef (w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var params = mux.Vars(r)
+
```

Comment:
We should be returning 404 if the survey doesn't exist - do a select before doing the delete to make sure it exists